nixos/modules/services/networking/gns3.nix

```diff
@@ -0,0 +1,261 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.services.gns3;
+
+  configFile = pkgs.writeText "gns3-server.conf" ''
+    [Server]
+    appliances_path = /var/lib/gns3/appliances
+    configs_path = /var/lib/gns3/configs
+    images_path = /var/lib/gns3/images
+    projects_path = /var/lib/gns3/projects
+    symbols_path = /var/lib/gns3/symbols
+
+    ${lib.optionalString (cfg.ubridge.enable) ''
+    ubridge_path = ${cfg.ubridge.package}/bin/ubridge
+    ''}
+
+    auth = ${if cfg.auth.enable then "True" else "False"}
+    ${lib.optionalString (cfg.auth.user != null) ''
+      user = ${cfg.auth.user}
+    ''}
+    ${lib.optionalString (cfg.auth.password != null) ''
+      password = ${cfg.auth.password}
+    ''}
+
+    ${lib.optionalString (cfg.vpcs.enable) ''
+    [VPCS]
+    vpcs_path = ${cfg.vpcs.package}/bin/vpcs
+    ''}
+
+    ${lib.optionalString (cfg.dynamips.enable) ''
+    [Dynamips]
+    dynamips_path = ${cfg.dynamips.package}/bin/dynamips
+    ''}
+  '';
+
+in {
+  meta.maintainers = [ lib.maintainers.anthonyroussel ];
+
+  options = {
+    services.gns3 = {
+      enable = lib.mkOption {
+        type = lib.types.bool;
+        default = false;
+        description = lib.mdDoc ''Whether to enable GNS3 Server daemon.'';
+      };
+
+      package = lib.mkPackageOptionMD pkgs "gns3-server" {};
+
+      auth = {
+        enable = lib.mkOption {
+          type = lib.types.bool;
+          default = false;
+          description = lib.mdDoc ''Enable password based HTTP authentication to access the GNS3 server.'';
+        };
+
+        user = lib.mkOption {
+          type = lib.types.nullOr lib.types.str;
+          default = null;
+          example = "gns3";
+          description = lib.mdDoc ''Username used to access the GNS3 server.'';
+        };
+
+        password = lib.mkOption {
+          type = lib.types.nullOr lib.types.str;
+          default = null;
+          description = lib.mdDoc ''
+            Password used to access the GNS3 server.
+
+            Warning: do not set confidential information here
+            because it is world-readable in the Nix store.
+          '';
+        };
+      };
+
+      host = lib.mkOption {
+        type = lib.types.str;
+        default = "127.0.0.1";
+        description = lib.mdDoc ''IP address on which GNS3 Server will listen on.'';
+      };
+
+      port = lib.mkOption {
+        type = lib.types.port;
+        default = 3080;
+        description = lib.mdDoc ''Port on which GNS3 Server will listen on.'';
+      };
+
+      log = {
+        file = lib.mkOption {
+          type = lib.types.nullOr lib.types.path;
+          default = null;
+          example = "/var/log/gns3/server.log";
+          description = lib.mdDoc ''
+            Path of the file GNS3 Server should log to.
+            If `null`, it logs to stdout, which is not recommended by upstream.
```

Comment:
Hm, why is it not recommended by upstream? We often redirect logs to stdout because journald will pick it up.

Reply:
(please add a reference to this)

Reply:
I removed the line, it was a copy-paste error from a description block of another module.